### DISCUSSION TITLE
Guardian CGM No data available when using the Companion app

### DISCUSSION THREAD

**Discussion by qq254344379** [NightscoutFoundation/xDrip#2842] [General]
Hello, I am currently using Guardian CGM (3rd generation). Following the instructions, changing the Companion app to synchronize blood sugar to Xdrip was unsuccessful. What should I do to set it up? I do not want to use the network because my location has high network latency.
![bb34d692-ad25-41bc-99c6-1d62561629a7](https://github.com/NightscoutFoundation/xDrip/assets/65702480/050ee0c9-68c0-4712-8d05-0868b4036a0c)

**Comment by Navid200** [comment]
Does Guardian CGM show notifications on your Android phone?
Can you post a screenshot?

**Comment by qq254344379** [comment]
![mmexport1684934586938](https://github.com/NightscoutFoundation/xDrip/assets/65702480/aca94c8c-817d-406d-9cf2-37a987690a76)
![mmexport1684934583758](https://github.com/NightscoutFoundation/xDrip/assets/65702480/ab555b03-4914-4879-98b6-58c42b27a23b)
![Uploading mmexport1684934580817.jpg…]()

**Comment by Navid200** [comment]
It looks like xDrip is showing readings.  Isn't it?
The xDrip notification is your screenshot.

Can you make sure you have enabled the "Companion Bluetooth" setting?
I have explained the details here:  https://navid200.github.io/xDrip/docs/Follow/CompanionApp.html

**Reply by jamorham** [reply]
@Navid200 I don't think companion bluetooth is suitable for guardian sensor

**Reply by Navid200** [reply]
OK, thanks

**Reply by qq254344379** [reply]
![image](https://github.com/NightscoutFoundation/xDrip/assets/65702480/6c2c36ad-943d-4ba0-8ebf-537cf6ceea55)

Hello, I set it according to this setting

**Reply by Navid200** [reply]
Yes, you did.  But, you are not answering my question.  Here I am going to ask it for the third time:
Do you not have readings shown on the xDrip main screen?

**Reply by qq254344379** [reply]
not have

**Reply by Navid200** [reply]
Where is this xDrip reading coming from?
![240618458-aca94c8c-817d-406d-9cf2-37a987690a76](https://github.com/NightscoutFoundation/xDrip/assets/51497406/511da03f-6e5a-4dcc-b1cf-cfc6bda5083a)

**Reply by qq254344379** [reply]
CareLink follower

**Reply by Navid200** [reply]
If you want to use Companion mode, you need to select Companion app under hardware data source instead of CareLink.
You need to make sure that the companion app, in this case, Guardian, shows a new notification for every reading.  Some apps need some setting to be enabled for this to happen.
You need to choose the same glucose setting in xDrip and the companion app.
You need to be using at least the current stable release, currently, May 3, 2023.

If you do all of that, and you get no readings in xDrip, that would be unexpected.

**Comment by Navid200** [comment]
@qq254344379 
Does xDrip not show readings on its main screen?

**Reply by qq254344379** [reply]
Using CareLink follower is fine, but the network latency in my area is too high and there may be frequent interruptions.

**Reply by qq254344379** [reply]
Companion app No data

**Reply by jamorham** [reply]
Check xDrip units mmol/mgdl is same as your companion app. Please check xDrip logs in case there is any report of an unknown package which is not one seen before. Make sure xDrip has notification access.
